content/applications/finance/fiscal_localizations/kenya.rst

```diff
@@ -7,114 +7,320 @@ Kenya
-confirming the connection between the device and your computer.
-
-.. image:: kenya/iot-box.png
-   :align: center
-   :alt: Your IoT box is up and running
+Install the 🇰🇪 **Kenyan** fiscal localization package :ref:`fiscal localization package
+<fiscal_localizations/packages>` to get all the features of the Kenyan localization.
+
+eTIMS
+=====
+
+The `Kenya Revenue Authority (KRA) <https://www.kra.go.ke/>`_ has decided to go digital for tax
+collection through the `electronic Tax Invoice Management System (eTIMS)
+<https://www.kra.go.ke/online-services/etims>`_.
+
+Odoo has developed an :abbr:`OSCU (Online Sales Control Unit)` that integrates with the existing
+**Trader Invoicing System (TIS)**. The OSCU validates, encrypts, signs, transmits, and stores tax
+invoices.
 
 .. note::
-   If the device is not detected, try to plug it in again or click on the :guilabel:`Restart` button
-   in the top right corner.
+   Make sure to install the **Kenya eTIMS EDI** modules to fully use the OSCU device.
+
+OSCU device initialization
+--------------------------
+
+Users must initialize their OSCU before they can use it. Before initialization, navigate to
+:guilabel:`Odoo Database --> Settings --> General Settings`, click :guilabel:`Update Info`, and in
+the :guilabel:`General Information` tab, enter your tax ID in the :guilabel:`Tax ID` field.
+
+To initialize the OSCU, go to the :guilabel:`eTIMS` tab. Fill in the :guilabel:`eTIMS Branch Code`
+and :guilabel:`Serial Number` of the device. Make sure to set :guilabel:`eTIMS Server Mode` to
+:guilabel:`Test` for the initialization. When done, tick the **user agreement** and click
+:guilabel:`Initialize OSCU`. Your device is now authenticated.
+
+.. note::
+   Three :guilabel:`Server Mode` are available:
+
+   - :guilabel:`Demo`: Uses mocked data and does not require an initialized OSCU. Serves demo
+     purposes;
+   - :guilabel:`Test`: Used to test the connection to eTIMS;
+   - :guilabel:`Production`: Used for live databases ready to send data.
+
+.. Important::
+   If your device has **already been authenticated** (through another ERP, for example), enable
+   :doc:`../../general/developer_mode` and go to the :guilabel:`eTIMS` tab. The field
+   :guilabel:`Device Communication Key` is now available. Enter the key obtained through a previous
+   authentication, :guilabel:`Save manually`, and click :guilabel:`Initialize OSCU`. It **may take
+   time** before the button is available as the initialization was made elsewhere before.
+
+eTIMS codes
+-----------
+
+Common standard codes are **automatically** fetched from the KRA eTIMS API servers every **two
+days**. These can be fetched **manually** by enabling :doc:`../../general/developer_mode` and going
+to :menuselection:`Settings --> Technical --> Automation: Scheduled Actions -->
+KE eTIMS: Fetch KRA standard codes`. Here, click :guilabel:`Run Manually` to fetch the codes.
```

Comment:
I cleared up the menu path a bit. There must have been a runbot issue, because the "standard codes" can be found in it (see attached).
![standard codes](https://github.com/odoo/documentation/assets/104567387/87b3b643-1e0c-4453-8601-2bfdeef73599)